File: backends/java/zappy-domain/src/main/java/com/zappymart/domain/ordering/OrderPlaced.java
package com.zappymart.domain.ordering;

import com.zappymart.domain.shared.DomainEvent;

import java.time.Instant;

public record OrderPlaced(
        String orderId,
        String orderNumber,
        String customerId,
        String promotionCode,
        Instant occurredAt) implements DomainEvent {
}

File: backends/kotlin/zappy-domain/src/main/kotlin/nl/zappymart/domain/ordering/OrderPlaced.kt
package nl.zappymart.domain.ordering

import java.time.Instant
import nl.zappymart.domain.shared.DomainEvent
import nl.zappymart.domain.shared.PromotionCode

data class OrderPlaced(
    val orderId: String,
    val orderNumber: String,
    val customerId: String,
    val promotionCode: PromotionCode?,
    val placedAt: Instant,
) : DomainEvent
